Linking.md

```diff
@@ -262,30 +288,35 @@ offset of the immediate within the code section.
 The same technique applies for all function calls whether the function is
 imported or defined locally.
 
+The Wasm object files shall have imports corresponding to all undefined
+symbols, and all defined symbols have an associated export, including
+those with hidden visibility or internal (local) linkage.
+
+When taking the address of a non-existent function, a null index shall
+be written into the code/data section (for example, taking the address of
+a weak symbol which is not available at link time).
+
+XXX what about calling a weak symbol that isn't available? What can you
+do to make the resulting Wasm file valid? Bother, do we have to do something
+silly like create a stub function with the right signature that just calls
+unreachable?
+
+
 Merging Data Sections
 ---------------------
 
+Data symbols (C/C+ globals) are represented in the object file as Wasm
+data segments with an associated DATA symbol.
+
+In the final linked binary all the segments are merged, and relocations are
+used to update the code to read/write to the correct addresses.
```

Comment:
All segments with the same name prefix are merged.  i.e. .data.* segments all get merged, but they don't get merges with .rodata.*